sys/aarch64/pmap.c

```diff
@@ -76,7 +76,9 @@ pte_t pte_make(paddr_t pa, vm_prot_t prot, unsigned flags) {
 }
 
 pte_t pte_protect(pte_t pte, vm_prot_t prot) {
-  return vm_prot_map[prot] | (pte & (~ATTR_AP_MASK & ~ATTR_XN));
+  return vm_prot_map[prot] |
+         (pte & (~ATTR_AP_MASK & ~ATTR_XN & ~ATTR_SW_FLAGS)) |
+         (pte & ATTR_AP_USER); /* This flag shouldn't be cleaned */
```

Comment:
What do you mean by 'follow pattern from prevoius line'? This flag can't be cleaned so if it is set then we must be sure that it is set in result pte (it may be cleared with `ATTR_AP_MASK`).